src/v/pandaproxy/client/fetch_session.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "kafka/types.h"
+#include "model/fundamental.h"
+
+#include <absl/container/flat_hash_map.h>
+
+namespace pandaproxy::client {
+
+class fetch_session {
+public:
+    kafka::fetch_session_id id{kafka::invalid_fetch_session_id};
+    kafka::fetch_session_epoch epoch{kafka::initial_fetch_session_epoch};
+    absl::flat_hash_map<
```

Comment:
flat_hashes have a huge memory overhead and the seastar buddy allocator does very poorly w/ them. also, looks like what matters is the nested hash map. 

I'd make the first a node-map and the second, i would set the reallocation step at 100% of capacity. 

would be nice if this class actually hid the implementation of this nested map tbh. 

also, every time i'd add a partition, i would `shrink_to_fit` on the hash-map to actually get the benfit since we can observe 2 things. 

1. partitions are small
2. once we reach partition-size stability, it will never change. 

so because of this i suggest always srhink_to_fit() on this nested flat-map (i'd bet we get ok boost on perf too)